ol/genesis-tools/src/fork_genesis.rs

```diff
@@ -83,19 +89,33 @@ pub fn get_baseline_genesis_change_set(
 pub fn append_genesis(
     gen_cs: ChangeSet,
     legacy_vec: Vec<LegacyRecovery>,
+    expected_len_all_users: u64,
 ) -> Result<Transaction, Error> {
     // merge writesets
     let mut all_writesets = gen_cs.write_set().to_owned().into_mut();
     let mut total_coin_value = 0u64;
+    let mut len = 0u64;
+    // iterate through all the accounts
+    // this includes the Validator accounts.
+    // at a minimum balances will need to be recovered
+    // and the total tokens calculated, and checked for matching sums.
+    // there may also be migrations on other account state that may need to be done.
     for l in &legacy_vec {
+        assert!(l.account.is_some());
+        println!("migrating: {} - {}", &l.account.unwrap(), &len);
```

Comment:
Maybe `diem_logger::info` or `debug` here?

Reply:
Yeah would be good to use a logger. Good catch. We've often run into issues with using diem_logger externally. Will test.